lib/client.js

```diff
@@ -909,12 +909,18 @@ Request.prototype.end = function(fn){
   };
 
   // progress
+  var handleProgress = function(e){
+    if (e.total > 0) {
+      e.percent = e.loaded / e.total * 100;
+    }
+    self.emit('progress', e);
```

Comment:
What if `total < 0` ? We won't have `e.percent`. Do we want to have a default value or stick with undefined?